lib/tic_tac_toe.ex

```diff
@@ -10,25 +10,27 @@ defmodule TicTacToe do
     welcome = @welcome_message <> "\n" <> communicator_formatter.format_board(board)
     communicator.display(welcome)
 
-    player_move = String.trim(communicator.read_input())
-    board = Board.place_a_symbol(board, player_move, "X")
-    communicator.display(communicator_formatter.format_board(board))
+    player_input = String.trim(communicator.read_input())
+    case Validator.validate(player_input) do
+      {:ok, player_move} ->
+        board = Board.place_a_symbol(board, player_move, "X")
+        communicator.display(communicator_formatter.format_board(board))
 
-    board
-    |> get_game_status_message()
-    |> communicator.display()
+        board
+        |> get_game_status_message()
+        |> communicator.display()
+      {:error, _} ->
+        communicator.display("Invalid input, please input a number between 1 and 9\n")
+        start(communicator, communicator_formatter, board)
```

Comment:
I feel you and completely understand, it's been tough for me to think about/remember, I'm not used to no real iteration/loops 😂 Full transparency, I was a little nervous answering because I wasn't one hundred percent sure, I appreciate all the help and replies! 🥳

Reply:
Another good point, the player would not want to see the welcome message multiple times or any time after the first move for that matter. Right now they do see it every call, but this is a change I was thinking of implementing during the change player ticket as that is when the full game loop will be added. Do you think this is something I should address in this PR? Either way is good with me, regardless I agree it will be changed to not display multiple times

Reply:
I am looking into how I might go about not displaying the welcome message every time and I had a question, would the welcome message want to be avoided after the first call or after the first move? IE: If a user submits an invalid input, should the welcome message still be displayed if that invalid input was the first move (since the board is still empty) or should it only be displayed once no matter what?

Reply:
Thanks for clarifying! Yea I think it would be confusing if the user sees the welcome message after the game has started because it gives a false sense of the game is starting over from the beginning even though in reality it's not. In terms of addressing this now vs in future PRs. I don't have a preference so up to you.

Reply:
Would it be ok to have a flag/boolean passed into start that is defaulted as true and then every subsequent call of start after it is input as false? IE: `def start(communicator, communicator_formatter, board \\ Board.setup_initial_board(), game_started // false)` and pass it true in game_started whenever called after. 
I feel like there is a better way to go about this though, what are your thoughts? Thank you for all the help!

Reply:
Either way, I think this would be better in the change player turn ticket, so I will probably implement it then. Just trying to get a feel for how I might do it later, thank you again 😄